effektif-workflow-impl/src/main/java/com/effektif/workflow/impl/bpmn/BpmnWriterImpl.java

```diff
@@ -228,9 +231,38 @@ protected void writeWorkflow(AbstractWorkflow workflow) {
     workflow.writeBpmn(this);
     writeScope();
     endElement();
+    fixDiagramDuplicateIds(workflow);
     writeDiagram(workflow);
   }
 
+  /**
+   * Temporary #2932 fix - ensures that the export doesn’t generate XML that contains duplicate IDs (not well-formed).
+   * The fix is to preprend duplicate shape IDs
+   */
+  private void fixDiagramDuplicateIds(AbstractWorkflow workflow) {
+    Diagram diagram = workflow.getDiagram();
+    if (diagram != null) {
+      Set<String> activityIds = workflow.getActivities() == null ? Collections.emptySet() :
+        workflow.getActivities().stream().map(activity -> activity.getId()).collect(Collectors.toSet());
+
+      if (diagram.canvas.hasChildren()) {
+        diagram.canvas.children.stream()
+          .filter(shape -> activityIds.contains(shape.id))
+          .forEach(shape -> shape.id("shape-" + shape.id));
+      }
+
+      if (diagram.edges != null) {
+        diagram.edges.stream()
+          .filter(edge -> activityIds.contains(edge.fromId))
+          .forEach(edge -> edge.fromId("shape-" + edge.id));
```

Comment:
oops, that should be `"shape-" + edge.fromId` - same copy-paste error a few lines below